libnavigation-core/src/test/java/com/mapbox/navigation/core/fasterroute/FasterRouteTrackerTest.kt

```diff
@@ -0,0 +1,92 @@
+@file:OptIn(ExperimentalMapboxNavigationAPI::class)
+
+package com.mapbox.navigation.core.fasterroute
+
+import com.mapbox.navigation.base.ExperimentalMapboxNavigationAPI
+import com.mapbox.navigation.core.fasterroute.TestRoutes.Companion.FASTER_ROUTE_IN_MUNICH
+import com.mapbox.navigation.core.fasterroute.TestRoutes.Companion.MUNICH_NUREMBERG
+import com.mapbox.navigation.testing.LoggingFrontendTestRule
+import kotlinx.coroutines.runBlocking
+import org.junit.Assert.assertEquals
+import org.junit.Rule
+import org.junit.Test
+
+class FasterRouteTrackerTest {
+
+    @get:Rule
+    val loggerRule = LoggingFrontendTestRule()
+
+    @Test
+    fun `no faster route is available from Munich to Nuremberg moving by the slowest road`() =
+        runBlocking<Unit> {
+            val fasterRoutes = createFasterRoutesTracker()
+            val recordedRoutesUpdates = readRouteObserverResults(MUNICH_NUREMBERG)
+            for (recordedUpdate in recordedRoutesUpdates) {
+                val result = fasterRoutes.routesUpdated(
+                    recordedUpdate.update,
+                    recordedUpdate.alternativeMetadata.values.toList()
+                )
+                val alternativesFromUpdate = recordedUpdate.alternativeMetadata.values
+                    .map { it.alternativeId }
+                    .joinToString(separator = ",") { it.toString() }
+                assertEquals(
+                    "incorrect result for update with alternatives $alternativesFromUpdate",
+                    FasterRouteResult.NoFasterRoad,
+                    result
+                )
+            }
+        }
+
+    @Test
+    fun `faster route is available driving in Munich after picking the slowest road`() =
+        runBlocking<Unit> {
+            val fasterRoutesTracker = createFasterRoutesTracker()
+            val routeUpdates = readRouteObserverResults(FASTER_ROUTE_IN_MUNICH)
+            val fasterRoutesIds = mutableListOf<String>()
+            for (recordedUpdate in routeUpdates) {
+                val result = fasterRoutesTracker.routesUpdated(
+                    recordedUpdate.update,
+                    recordedUpdate.alternativeMetadata.values.toList()
+                )
+                if (result is FasterRouteResult.NewFasterRoadFound) {
+                    fasterRoutesIds.add(result.route.id)
+                }
+            }
+            assertEquals(
+                listOf(
+                    "JJXKpAxo3Yhh3rbJJmToNPPkzjh-hJHZV8u2Uksl1gwue_3sdZZ0ig==#1",
+                    "b_dZe9Trx9MkOIphyCILePn4cBI6taAQmSctw1k5jaNzWz8vL-10-w==#1",
+                    "b_dZe9Trx9MkOIphyCILePn4cBI6taAQmSctw1k5jaNzWz8vL-10-w==#1"
+                ),
+                fasterRoutesIds
+            )
+        }
```

Comment:
Now I realise that it's hard to read those tests because routes ids don't say much about scenario. I recorded seeing the map and routes, but without it reader has to manually check geometries of the routes to say if test has a valid scenario. Maybe if I put links to geometries decoder it helps 🤔